meson-scripts/build_libbpf

```diff
@@ -7,18 +7,27 @@ out=${out#\"}
 out=${out%\"}
 args=($out)
 
+declare -a cflags=()
+
 idx=0
 cc=${args[idx]}
 if [ "$cc" = "ccache" ]; then
     idx=$((idx+1))
     cc="$cc ${args[idx]}"
 fi
 
-declare -a cflags=()
+# This fixes an issue that was happening in some cases when building with gcc:
+# relocation R_X86_64_32 against `.rodata' can not be used when making a PIE
+# object; recompile with -fPIE
+if [ "$cc" = "cc" ]; then
+    cflags+="-fPIE "
+fi
+
+cflags+="-O2 "
 
 for arg in ${args[@]:(idx+1)}; do
     case $arg in
-	-I*|-M*|-o|-c) ;;
+	-I*|-M*|-O*|-o|-c) ;;
 	-*) cflags+="$arg ";;
     esac
 done
```

Comment:
By the way, not sure it matters but I believe we should only need to build `libbpf.a` target

Reply:
Is there a reason for excluding -O? Why don't we want to be synced with how other parts of scx is built?